commons/all/src/main/java/org/infinispan/commons/configuration/attributes/AttributeDefinition.java

```diff
@@ -82,6 +90,14 @@ public boolean isGlobal() {
       return global;
    }
 
+   public boolean isDeprecated() {
+      return (deprecatedMajor < Integer.MAX_VALUE);
+   }
+
+   public boolean isDeprecated(int major, int minor) {
```

Comment:
Neither `isDeprecated` method appears to be called. Do you have ause-case in mind for these?

Reply:
I've fixed `ParseUtils.parseAttributes()` to report use of deprecated attributes. Now we just need to decorate those we want to deprecate :)